### PR TITLE
adjust sync-config CI workflow

### PR DESCRIPTION
**Pull request by hidalgopl**
enhancement: sync vcluster-config on beta, rc and stable versions; sync vcluster-docs main version on beta and rc releases

**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves ENG-3393

**Please provide a short message that should be published in the vcluster release notes**
[Fixed an issue where vcluster ...](enhancement: sync vcluster-config on beta, rc and stable versions; sync vcluster-docs main version on beta and rc releases)

**What else do we need to know?** 
Based on the discussions, we want to sync docs on beta & rc versions and sync vcluster-config on beta, rc and stable versions.

### CODE REVIEW

File: .github/workflows/sync-config-schema.yaml [hunk 1]
```diff
@@ -30,20 +31,33 @@ jobs:
...
+
+      - name: Skip sync on alpha
+        if: ${{ steps.release.outputs.is_alpha_version == 'true' }}
+        env:
+          RELEASE_TAG: ${{ steps.release.outputs.release_tag }}
+        run: echo "skipping sync because release ${RELEASE_TAG} is alpha"
 
       - name: Checkout repo
-        if: ${{ steps.release.outputs.update_main == 'false' }}
+        if: ${{ steps.release.outputs.is_alpha_version == 'false' }}
```
Comment (rohantmp) [inline .github/workflows/sync-config-schema.yaml:53]: `steps.release.outputs.is_stable_version == 'false' && && steps.release.outputs.is_alpha_version == 'false' `?

File: .github/workflows/sync-config-schema.yaml [hunk 2]
```diff
@@ -53,13 +67,14 @@ jobs:
           GH_ACCESS_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
 
       - name: Set up Go
-        if: ${{ steps.release.outputs.update_main == 'false' }}
+        if: ${{ steps.release.outputs.is_alpha_version == 'false' }}
```
Comment (rohantmp) [inline .github/workflows/sync-config-schema.yaml:70]: `steps.release.outputs.is_stable_version == 'false' && && steps.release.outputs.is_alpha_version == 'false' `?

File: .github/workflows/sync-config-schema.yaml [hunk 3]
```diff
@@ -114,9 +129,11 @@ jobs:
 
 
       - name: Update vcluster schema in vcluster-config
-        if: ${{ steps.release.outputs.update_main == 'false' }}
+        # update only on beta, -rc and stable versions
+        if: ${{ steps.release.outputs.is_alpha_version == 'false' }}
```
Comment (rohantmp) [inline .github/workflows/sync-config-schema.yaml:133]: `steps.release.outputs.is_stable_version == 'false' && && steps.release.outputs.is_alpha_version == 'false' `?

**Review by rohantmp — CHANGES_REQUESTED** [review]
Looks good, minor fixes requested

File: .github/workflows/sync-config-schema.yaml [hunk 4]
```diff
@@ -53,13 +67,14 @@ jobs:
...
         uses: actions/setup-go@v5
         with:
           go-version-file: go.mod
 
-      - name: Update main docs version on alpha or beta versions
-        if: ${{ steps.release.outputs.update_main == 'true' }}
+      - name: Update main docs version on beta or rc versions
+        # update docs "main" version only on beta or -rc
+        if: ${{ steps.release.outputs.is_stable_version == 'false' && steps.release.outputs.is_alpha_version == 'false' }}
```
Comment (rohantmp) [inline .github/workflows/sync-config-schema.yaml:77]: the other conditions should be like this no?

Comment (hidalgopl) [reply .github/workflows/sync-config-schema.yaml:77]: no. vCluster docs are now versioned, so we have "main" version (for the latest beta / RC version) and a list of stable version where each one has dedicated docs (e.g. `v0.19`, `v0.20`, `v0.21`).
We update docs "main" version only on beta & RC versions. For full versions, there is a CI job in `vcluster-config` repository that updates the given stable version in docs.

File: .github/workflows/sync-config-schema.yaml (hunk 3 above)
Comment (hidalgopl) [reply .github/workflows/sync-config-schema.yaml:133]: no, we want to do this on betas, RC and full versions (see my comment below)

File: .github/workflows/sync-config-schema.yaml (hunk 2 above)
Comment (hidalgopl) [reply .github/workflows/sync-config-schema.yaml:70]: no, we want to do this on betas, RC and full versions (see my comment below)

File: .github/workflows/sync-config-schema.yaml (hunk 1 above)
Comment (hidalgopl) [reply .github/workflows/sync-config-schema.yaml:53]: no, we want to do this on betas, RC and full versions (see my comment below)

**Review by rohantmp — APPROVED** [review]

File: .github/workflows/sync-config-schema.yaml (hunk 4 above)
Comment (rohantmp) [reply .github/workflows/sync-config-schema.yaml:77]: Makes sense! I misunderstood the implications of the difference!

**Pull request merged**